libs/cases/frontend/view/src/lib/components/detail-case/detail-case.component.ts

```diff
@@ -27,5 +30,135 @@ export class DetailCaseComponent {
         return _case.find((_case) => String(_case.id) === this.route.snapshot.params['id']);
     });
 
-    constructor(private route: ActivatedRoute) {}
+    isEditing = false;
+    wasValidated = false;
+    datePattern = /^\d{4}-\d{2}-\d{2}T00:00:00\.000Z$/;
+
+
+    constructor(private route: ActivatedRoute, private readonly _modalService: ModalService, private readonly toastService: ToastService) {
+    }
+
+    deleteCase() {
+        const caseId = this.mapCaseId(this.casedetail());
+        if (caseId !== undefined) {
+            this._browseFacade.deleteCase(caseId).subscribe();
+        }
+    }
+
+    mapCaseId(_case: ICaseResponse | undefined) {
+        if (_case === undefined) {
+            return undefined;
+        }
+        return {
+            id: _case.id,
+        };
+    }
+
+    async deleting() {
+        const instance = await this._modalService.open({
+            content: DeleteModalComponent,
+        });
+
+        // modal closes on confirm and dismisses on cancel
+        instance.onClose.on(() => {
+            this.deleteCase();
+        });
+    }
+
+    enableEditing() {
+        this.isEditing = true;
+    }
+
+    cancelEdit() {
+        this.isEditing = false;
+    }
+
+    toggleEdit() {
+        if (this.isEditing) {
+            this.onSubmit();
+        } else {
+            this.isEditing = true;
+        }
+    }
+
+    onSubmit(): void {
+        // TODO validate form
+        if (this.validateForm() === 'valid') {
+            const caseId = this.mapCaseId(this.casedetail());
+            const caseData = this.casedetail();
+
+            if (caseId !== undefined && caseData !== undefined) {
+                this._browseFacade.updateCase(caseId, caseData).subscribe({});
+            }
+            this.isEditing = false;
+        } else {
+            this.showErrorToast(this.validateForm());
+        }
+    }
+
+    validateForm() {
+        const casedetail = this.casedetail();
+
+        if (casedetail !== undefined) {
+
+            if (this.casedetail()?.title === '') {
+                return 'Empty title';
+            }
+
+
+            if (casedetail.status !== undefined) {
+                if (!Object.values(ECaseStatus).includes(casedetail.status)) {
+                    return 'Status is not in List: OPEN, INPROGRESS, OVERDUE, ONHOLD, DONE, CANCELLED, ARCHIVED';
+                }
+            }
+            if (casedetail.priority !== undefined) {
+            if (!Object.values(ECasePriority).includes(casedetail.priority)) {
+                return 'Priority is not in List: EMERGENCY, HIGH, MEDIUM, LOW';
+            }}
```

Comment:
these brackets are weird (also in the lines after this). Press `CRTL ALT L` to format everything

Reply:
actually I think the first if 
```
if (casedetail.priority !== undefined) {
```

might actually be unnecessary

Reply:
I did change that, this if was indeed not necessary in the other spots as well, but better safe than sorry